cpp/src/arrow/util/io_util.cc

```diff
@@ -1628,6 +1629,8 @@ Result<int64_t> FileRead(int fd, uint8_t* buffer, int64_t nbytes) {
   HANDLE handle = reinterpret_cast<HANDLE>(_get_osfhandle(fd));
 #endif
   int64_t total_bytes_read = 0;
+  util::tracing::Span span;
+  START_SPAN(span, "FileRead", {{"fd", fd}});
```

Comment:
This is very deep in the I/O & filesystem hierarchy. Maybe it would be better to put these spans in `LocalFileSystem` instead?